presidio-anonymizer/presidio_anonymizer/entities/anonymizer_request.py

```diff
@@ -71,40 +71,43 @@ def __handle_analyzer_results(self, data):
             raise InvalidParamException(
                 "Invalid input, " "analyzer results can not be empty"
             )
+        text_len = len(data.get("text"))
         for analyzer_result in analyzer_results:
-            self._analysis_results.append(AnalyzerResult(analyzer_result))
+            analyzer_result = AnalyzerResult(analyzer_result)
+            analyzer_result.validate_position_in_text(text_len)
+            self._analysis_results.append(analyzer_result)
 
-    def __handle_transformations(self, data):
+    def __handle_anonymizers(self, data):
         """
-        Go over the transformations and get the relevant anonymizer class for it.
+        Go over the anonymizers and get the relevant anonymizer class for it.
 
-        Inserts the class to the transformation so the engine will use it.
-        :param data: contains the text, transformations and analyzer_results
+        Inserts the class to the anonymizer so the engine will use it.
```

Comment:
nit: Not directly related to this PR
```suggestion
        Inserts the requested anonymizer into the list of anonymizers so the engine will use it.
```
Is this what this this function does? A little difficult to understand based on the method name and docstring.

Reply:
This entire class is going to be gone soon... 
so lets leave it as is for now :)